### ISSUE TITLE
question - how to add custom set of icons (f.e. flaticons)

### ISSUE THREAD

**Issue by Jimmi08** [e107inc/e107#1739]
In menus I need to use custom flaticons set. 
example:

```
                       <!-- ICON BLOCK -->
                        <div class="icon-block">
                            <div class="icon-container">
                                <a href="#"><i class="flaticon-brush20"></i></a>
                            </div>
                            <div class="icon-text">
                                <h3>Painting</h3>
                                <p>Voluptate illum dolore ita ipsum, quid deserunt singulis, labore admodum ita multos malis ea nam nam tamen fore amet. Vidisse quid incurreret ut ut possumus.</p>
                            </div>
                        </div>
```

Now I am using menu_button_text for this with theme shortcode, but it would be nice to insert name of icon in icon field (something like next option tab in media manager). 

```
                        <!-- ICON BLOCK -->
                        <div class="icon-block">
                            <div class="icon-container">
                                <a href="#"><i class="{MENU_BUTTON_TEXT}"></i></a>
                            </div>
                            <div class="icon-text">
                                <h3>{CMENUTITLE}</h3>
                                <p>{CMENUBODY}</p>
                            </div>
                        </div>  
```

Result:
![image](https://cloud.githubusercontent.com/assets/5429548/16384592/cb05c1c4-3c89-11e6-812a-dcb4bebe6ce9.png)

Is there other way how to solve this? Thanks

**Comment by rica-carv** [comment]
@Jimmi08 Can't you use css and try to put that icon like i'm doing with custom font asewome icons on my theme?
Like [this css file](https://github.com/rica-carv/e107_theme-selospt_v2/blob/88d83f456a3910640a470cabacbaa45f92d81229/fam-icons.css).
For instance, if i got in html something like this:
`<i class="fa fa-tags"></i> &nbsp;None &nbsp; <i class="fa fa-folder-open"></i> &nbsp;<a class="_category" style="" href="/news/list/category?">Misc</a>`, all fa & fa-xxxx class icons are rendered according to my css (as long it is available and loaded, of course...)

**Comment by Jimmi08** [comment]
Yes, but I need to do this dynamic, each menu has different icon, it needs to be changeable in admin area, not hardcoded.

**Comment by rica-carv** [comment]
Yes, i got your point.....
If you use in the custom menu Menu Icon/Glyph or Menu Image/Video, what is rendered???

**Comment by Jimmi08** [comment]
I can't render icon/glyph, so I render another free text field menu_button_text. Now for temporarily solving this.

**Comment by rica-carv** [comment]
Well, it seems i can...
![untitled 1](https://cloud.githubusercontent.com/assets/17130530/16388342/9de134ce-3c90-11e6-84cc-d05337ab7ee8.jpg)

**Comment by Jimmi08** [comment]
How you insert your icon via media manager?

**Comment by rica-carv** [comment]
In admin area, Content -> Pages / Menus -> Edit any menu -> On top, you see four tabs, select third tab: Menu. On the page bottom you see the options to Menu Icon/Glyph or Menu Image/Video...
LOL

**Comment by rica-carv** [comment]
Now, styling that like you wnat it's gonna be a headache, since that full line is [hardcoded, without templating](https://github.com/e107inc/e107/blob/2f9e14280dbf8bfcae6d26cb6c2afcec8c4862c8/e107_handlers/menu_class.php#L496)....

**Comment by Jimmi08** [comment]
I don't understand. I don't want to upload image. I want to insert text "flaticon-brush20"
![image](https://cloud.githubusercontent.com/assets/5429548/16388691/86a3ccd6-3c9a-11e6-82f5-acc0f9941a6e.png)

**Comment by rica-carv** [comment]
And where's that "flaticon-brush20"? In media manager? Does it has the correct category to show as icon?

**Comment by Jimmi08** [comment]
That was the original point of this question. I don't know how to add it there.
Flaticons are just fonts set similar like font-awesome fonts. 
Could we delete our discussion because nobody will read so long discussion? Thanks

**Comment by rica-carv** [comment]
@Jimmi08 Sorry, now i can sse REALLY what you need...
As far i see it, Glyphs would be what you need, but there's no where in Admin area to setup those Glyphs....

**Comment by Jimmi08** [comment]
Yes, it would be nice to have 4th tab and there insert icon as text field.

**Comment by rica-carv** [comment]
Or allow admin to define which Glyphs appear on the 3rd tab, Glyphs...
That would even be better, since it works if you pick it in admin area:
![untitled 2](https://cloud.githubusercontent.com/assets/17130530/16389587/87306a64-3c96-11e6-852f-32f48dc7419e.jpg)
![untitled 1](https://cloud.githubusercontent.com/assets/17130530/16389588/8731150e-3c96-11e6-80b0-2208ffc3afb9.jpg)

P.S.: Icon in image is custom, due to my css on the theme, but it would render the default Glyphicon, as you can see by the HTML....

**Comment by Jimmi08** [comment]
Generally it would be enough to have some free text fields for any purpose.

**Comment by CaMer0n** [comment]
This was solved and included in 2.1.4
